omod/src/main/java/org/openmrs/module/fhir2/narrative/OpenMRSNarrativeTemplateManifest.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.narrative;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import ca.uhn.fhir.context.ConfigurationException;
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.narrative.DefaultThymeleafNarrativeGenerator;
+import ca.uhn.fhir.narrative2.INarrativeTemplate;
+import ca.uhn.fhir.narrative2.INarrativeTemplateManifest;
+import ca.uhn.fhir.narrative2.TemplateTypeEnum;
+import com.google.common.base.Charsets;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.hl7.fhir.instance.model.api.IBase;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.openmrs.util.OpenmrsUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OpenMRSNarrativeTemplateManifest implements INarrativeTemplateManifest {
```

Comment:
Couldn't this just be something like:

```java
public class OpenMRSNarrativeTemplateManifest extends ca.uhn.fhir.narrative2.NarrativeTemplateManifest {
	public static NarrativeTemplateManifest forManifestFileLocation(String... propertyFiles) throws IOException {
		return forManifestFileLocation(Arrays.asList(propertyFiles));
	}

	public static NarrativeTemplateManifest forManifestFileLocation(Collection<String> propertyFiles) throws IOException {
		log.debug("Loading narrative properties file(s): {}", propertyFiles);
		return new NarrativeTemplateManifest(propertyFiles.stream().map(r -> loadResource(r)).map(p -> NarrativeTemplateManifest.loadProperties(p))
			.flatMap(p -> p.stream()).map(t -> new OpenmrsNarrativeTemplate(t)).collect(Collectors.toList()));
	}

	static String loadResource(String name) throws IOException {
		if (name.startsWith("openmrs:")) {
			File file = new File(OpenmrsUtil.getApplicationDataDirectory(), name.substring("openmrs:".length()));

			if (file.exists() == false) {
				throw new IOException("File not found: " + file.getAbsolutePath());
			}

			try (FileInputStream inputStream = new FileInputStream(file)) {
				return IOUtils.toString(inputStream, Charsets.UTF_8);
			}
		} else {
			return NarrativeTemplateManifest.loadResource(name);
		}
	}
}
```

(The above adds a small dependency that we create a constructor for `OpenmrsNarrativeTemplate` that takes a `NarrativeTemplate`, but that isn't too hard. That way we cut the code down to what we minimally need to support this change.

Reply:
Yeah actually we could do this. I didn't think of it this way initially because of the constructor. But yeah, this looks much better.

Reply:
@ibacher I just forgot about it, but `NarrativeTemplateManifest` cannot be extended because of a private constructor right?

Reply:
Yes... that's a good point. And I guess there isn't an easy way to remove a bunch of duplication here.

Reply:
@ibacher Hmm.. and since some methods of `NarrativeTemplateManifest` call some private methods of `NarrativeTemplate` class, we can't reduce that either. So what should be done instead?

Reply:
Are you sure there are methods in `NarrativeTemplateManifest` that call *private* methods of `NarrativeTemplate`? Because, by definition, that's not possible... If it's just a matter of package-private methods, we can always stick our version of `OpenMRSNarrativeTemplateManifest` in the `ca.uhn.fhir.narrative2` package.

The alternative is to dress these classes up so that they more closely resemble OpenMRS coding conventions: Setters should be `public` or at least `protected`. We should drop `the` and `my` as prefixes to variables, etc. I'd also like to minimize duplicating code as much as possible.

Reply:
> Are you sure there are methods in `NarrativeTemplateManifest` that call _private_ methods of `NarrativeTemplate`?

Ah, my bad. I meant non-public methods.

Reply:
> The alternative is to dress these classes up so that they more closely resemble OpenMRS coding conventions: Setters should be `public` or at least `protected`. We should drop `the` and `my` as prefixes to variables, etc. I'd also like to minimize duplicating code as much as possible.

Okay.